### ISSUE TITLE
[Bug]: Dependency versioning of platform packages

### ISSUE THREAD

**Issue by SvehlaJan** [Baseflow/flutter-geolocator#1344]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [X] iOS
- [X] Linux
- [X] macOS
- [X] Web
- [X] Windows

### Steps to reproduce

Define a specific (older) version of Geolocator in project's pubspec.yaml - ie. `geolocator: 9.0.2`.

### Expected results

The functionality and API of Geolocator remains unchanged until version in project's pubspec.yaml is changed.

### Actual results

Geolocator's `geolocator_platform_interface` sub-dependency (and other dependencies) are implicitly updated to the latest release - which breaks the API in this case.

### Code sample

`geolocator: 9.0.2`

### Version

9.0.2

### Flutter Doctor output

-

**Comment by jankuss** [comment]
We are also running into the same problem right now. We depend on `geolocator: ^9.0.2`

```

[2023-09-26T05:47:24.416Z] XXXXX.dart: Error: Required named parameter 'altitudeAccuracy' must be provided.
[2023-09-26T05:47:24.416Z] final Position userPosition = Position(
[2023-09-26T05:47:24.416Z]                                       ^
[2023-09-26T05:47:24.416Z] /root/.pub-cache/hosted/pub.dev/geolocator_platform_interface-4.1.0/lib/src/models/position.dart:9:9: Context: Found this candidate, but the arguments don't match.
[2023-09-26T05:47:24.416Z]   const Position({
[2023-09-26T05:47:24.416Z]         ^^^^^^^^
[2023-09-26T05:47:24.416Z] XXXX.dart: Error: Required named parameter 'altitudeAccuracy' must be provided.
```

So basically it boils down to this: Version `9.0.2` of `geolocator` requires `geolocator_platform_interface: ^4.0.3`. This resolves to version `4.1.0`, which contains breaking changes.

In our specific case, `Position` now has a new required attribute which was not there in versions `4.0.x` of `geolocator_platform_interface`.

**Comment by jankuss** [comment]
To resolve this, I believe the following needs to be done with the package to mitigate this issue:

- Publish a version `4.1.1` of `geolocator_platform_interface`, which reverts any breaking changes (this should just be the same code as `4.0.x`)
- Publish a version `5.0.0` of `geolocator_platform_interface` with the intended breaking changes.

Any future `flutter pub get` with `geolocator: ^9.0.0` will then resolve to `4.1.1` instead, which will not have any breaking changes. The `5.0.0` version can then be used for newer versions.
